src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -80,7 +86,34 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             // No AT in the cache or AT needs to be refreshed
             try
             {
-                return await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                if (cachedAccessTokenItem == null)
+                {
+                    return await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    if (cachedAccessTokenItem.NeedsRefresh())
+                    {
+                        _ = Task.Run(async () =>
+                        {
+                            // TODO - Exception handling is being discussed.
+                            try
+                            {
+                                await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                            }
+                            catch (MsalServiceException ex)
```

Comment:
Maybe we can do a little more here: 

MsalServiceException

- if the error code is 400 - this is most likely a config problem, e.g. a cert expired - log.Error
- if the Error Code is is 5xx (IsAadUnavailable) - log.Warning
- all other exceptions - log.Warning